src/analysis/bounded_variation.lean

```diff
@@ -753,6 +792,50 @@ begin
         evariation_on.Icc_add_Icc f xy yz ys], },
 end
 
+@[protected]
+lemma edist_zero_of_eq_zero
+  {f : α → E} {s : set α} (hf : has_locally_bounded_variation_on f s)
+  {a b : α} (ha : a ∈ s) (hb : b ∈ s) (h : variation_on_from_to f s a b = 0) :
+  edist (f a) (f b) = 0 :=
+begin
+  apply le_antisymm _ (zero_le _),
+  rw ←ennreal.of_real_zero,
+  rcases le_total a b with h'|h',
+  work_on_goal 2
+  { rw [eq_neg_swap, neg_eq_zero] at h,
+    rw [edist_comm],
+    swap_var [a b, ha ↔ hb], },
+  all_goals
```

Comment:
the goal management here is a little ugly. Can you do it with `wlog`?

Reply:
```
@[protected]
lemma edist_zero_of_eq_zero
  {f : α → E} {s : set α} (hf : has_locally_bounded_variation_on f s)
  {a b : α} (ha : a ∈ s) (hb : b ∈ s) (h : variation_on_from_to f s a b = 0) :
  edist (f a) (f b) = 0 :=
begin
  apply le_antisymm _ (zero_le _),
  rw ←ennreal.of_real_zero,
  wlog h' : a ≤ b := le_total a b using [b a, a b],
  { rw [←h, eq_of_le f s h', ennreal.of_real_to_real (hf a b ha hb)],
    apply evariation_on.edist_le,
    exacts [⟨ha, ⟨le_rfl, h'⟩⟩, ⟨hb, ⟨h', le_rfl⟩⟩], },
end
```
leaves the goal:
```
α: Type u_1
_inst_1: linear_order α
E: Type u_3
_inst_3: pseudo_emetric_space E
f: α → E
s: set α
hf: has_locally_bounded_variation_on f s
ba: α
h': b ≤ a
this: b ∈ s → a ∈ s → variation_on_from_to f s b a = 0 → edist (f b) (f a) ≤ ennreal.of_real 0
ha: a ∈ s
hb: b ∈ s
h: variation_on_from_to f s a b = 0
a_1: ℝ≥0
H: a_1 ∈ ennreal.of_real 0
⊢ ∃ (b_1 : ℝ≥0) (H : b_1 ∈ edist (f a) (f b)), b_1 ≤ a_1

```

Reply:
```lean
  wlog h' : a ≤ b := le_total a b using [b a, a b] tactic.skip,
  { apply le_antisymm _ (zero_le _),
    rw [←ennreal.of_real_zero, ←h, eq_of_le f s h', ennreal.of_real_to_real (hf a b ha hb)],
    apply evariation_on.edist_le,
    exacts [⟨ha, ⟨le_rfl, h'⟩⟩, ⟨hb, ⟨h', le_rfl⟩⟩] },
  { assume ha hb hab,
    rw edist_comm,
    apply this hb ha,
    rw [eq_neg_swap, hab, neg_zero] }
```

Reply:
You committed a version which is slightly different from mine, but mine is better :-)